tests/test_kernel.py

```diff
@@ -807,6 +873,52 @@ def basicTestComputeImageRaise(self, kernel, doRaise, kernelDescr=""):
                 self.fail(kernelDescr +
                           ".computeImage should not have raised an exception")
 
+    def compareResizedKernels(self, kernel1, kernel2):
+        """Compare two kernels' parameters and images where overlapping, but not size info
+        """
+        retStrs = []
+        if kernel1.isSpatiallyVarying() != kernel2.isSpatiallyVarying():
+            retStrs.append("isSpatiallyVarying differs: %s != %s" %
+                           (kernel1.isSpatiallyVarying(), kernel2.isSpatiallyVarying()))
+        if kernel1.getSpatialParameters() != kernel2.getSpatialParameters():
+            retStrs.append("spatial parameters differ: %s != %s" %
+                           (kernel1.getSpatialParameters(), kernel2.getSpatialParameters()))
+        if kernel1.getNSpatialParameters() != kernel2.getNSpatialParameters():
+            retStrs.append("# spatial parameters differs: %s != %s" %
+                           (kernel1.getNSpatialParameters(), kernel2.getNSpatialParameters()))
+        if not kernel1.isSpatiallyVarying() and hasattr(kernel1, "getKernelParameters"):
+            if kernel1.getKernelParameters() != kernel2.getKernelParameters():
+                retStrs.append("kernel parameters differs: %s != %s" %
+                               (kernel1.getKernelParameters(), kernel2.getKernelParameters()))
+        if retStrs:
+            return "; ".join(retStrs)
```

Comment:
I'm guessing original author of `compareKernels` wanted all the kernel differences to be printed out rather than the first to fail.  I agree It is a little unexpected in a unit test. I pulled out some of the duplication between the kernel comparison functions, but left in the string concatenation.